batcher/aligned-batcher/src/retry/batcher_retryables.rs

```diff
@@ -77,12 +85,78 @@ pub async fn get_gas_price_retryable(
     })
 }
 
+pub async fn create_new_task_retryable(
+    batch_merkle_root: [u8; 32],
+    batch_data_pointer: String,
+    proofs_submitters: Vec<Address>,
+    fee_params: CreateNewTaskFeeParams,
+    payment_service: &BatcherPaymentService,
+    payment_service_fallback: &BatcherPaymentService,
+) -> Result<TransactionReceipt, RetryError<BatcherError>> {
+    info!("Creating task for: 0x{}", hex::encode(batch_merkle_root));
+    let call_fallback;
+    let call = payment_service
+        .create_new_task(
+            batch_merkle_root,
+            batch_data_pointer.clone(),
+            proofs_submitters.clone(),
+            fee_params.fee_for_aggregator,
+            fee_params.fee_per_proof,
+            fee_params.respond_to_task_fee_limit,
+        )
+        .gas_price(fee_params.gas_price);
+
+    let pending_tx = match call.send().await {
+        Ok(pending_tx) => pending_tx,
+        Err(ContractError::Revert(err)) => {
+            // Since transaction was reverted, we don't want to retry with fallback.
+            warn!("Transaction reverted {:?}", err);
+            return Err(RetryError::Permanent(BatcherError::TransactionSendError));
+        }
+        _ => {
+            call_fallback = payment_service_fallback
+                .create_new_task(
+                    batch_merkle_root,
+                    batch_data_pointer,
+                    proofs_submitters,
+                    fee_params.fee_for_aggregator,
+                    fee_params.fee_per_proof,
+                    fee_params.respond_to_task_fee_limit,
+                )
+                .gas_price(fee_params.gas_price);
+            match call_fallback.send().await {
+                Ok(pending_tx) => pending_tx,
+                Err(ContractError::Revert(err)) => {
+                    warn!("Transaction reverted {:?}", err);
+                    return Err(RetryError::Permanent(BatcherError::TransactionSendError));
+                }
+                _ => return Err(RetryError::Transient(BatcherError::TransactionSendError)),
+            }
+        }
+    };
+
+    // timeout to prevent a deadlock while waiting for the transaction to be included in a block.
+    timeout(Duration::from_millis(BATCH_INCLUSION_DELAY), pending_tx)
```

Comment:
Shouldn't this return `TransientErrors` as they should be recoverable by sending the tx again.

Reply:
@MarcosNicolau if the timeout is reached, we should cancel the tx. This is because as the gas price might have changed, if we want to re-send the same transaction we should re-calculate what users want to have their proof included in the batch depending on the max fee they have set. 
As this is a very complicated logic, it is much easier to just cancel the transaction for now, so the timeout error should be a permanent one

Reply:
Oh yes!